pkgs/tools/backup/restic/default.nix

```diff
@@ -2,20 +2,21 @@
 
 buildGoPackage rec {
   name = "restic-${version}";
-  version = "0.9.2";
+  version = "0.9.3";
 
   goPackagePath = "github.com/restic/restic";
 
   src = fetchFromGitHub {
     owner = "restic";
     repo = "restic";
     rev = "v${version}";
-    sha256 = "0kl8yk636i3y7f2kd43pydjh4pv7hhq09p5k54jlysnrbf2kjb4h";
+    sha256 = "0l35pdrq1hhcz3cb2qm267m6846mxfwbl1adk2kp748b2q55pdma";
   };
 
+
   buildPhase = ''
     cd go/src/${goPackagePath}
-    go run build.go
+    GO111MODULE=on GOCACHE=`mktemp -d` go run -mod=vendor build.go
```

Comment:
Actually we already set `GOCACHE = "off"`, but this does not work?

Reply:
No, I'm getting either `build flag -mod=vendor only valid when using modules` or `go: cannot use modules with build cache disabled` if I I also remove `-mod=vendor`. It also doesn't work if I remove `GO111MODULE=on`.

Reply:
No, I'm getting either `build flag -mod=vendor only valid when using modules` or `go: cannot use modules with build cache disabled` if I I also remove `-mod=vendor`. It also doesn't work if I remove `GO111MODULE=on`.

Reply:
No, I'm getting either `build flag -mod=vendor only valid when using modules` or `go: cannot use modules with build cache disabled` if I I also remove `-mod=vendor`. It also doesn't work if I remove `GO111MODULE=on`.

Reply:
I wonder if we could detect go modules automatically.